TestMaybeRemove doesn't see removal in 	removedRng := <-fq.maybeRemovedRngs

The following test appears to have failed:

[#10176](https://circleci.com/gh/cockroachdb/cockroach/10176):

```
I1214 22:56:05.536495 971 raft/raft.go:487  [group 1] 1 became leader at term 6
I1214 22:56:05.537981 971 storage/replica_command.go:1209  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC +1.000s
I1214 22:56:05.538157 971 storage/replica.go:1368  gossiping system config from store 1, range 1
I1214 22:56:05.538629 971 storage/replica.go:1317  gossiping sentinel from store 1, range 1
I1214 22:56:05.538858 971 storage/replica.go:1323  gossiping first range from store 1, range 1
panic: test timed out after 5m0s

goroutine 2730 [running]:
testing.startAlarm.func1()
	/tmp/workdir/go/src/testing/testing.go:703 +0x16b
created by time.goFunc
	/tmp/workdir/go/src/time/sleep.go:129 +0x6e

goroutine 1 [chan receive, 4 minutes]:
testing.RunTests(0x1a00f40, 0x2094e20, 0xda, 0xda, 0x1)
	/tmp/workdir/go/src/testing/testing.go:562 +0xafa
--
	/go/src/github.com/cockroachdb/cockroach/storage/queue.go:287 +0x427
github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1(0xc820262360, 0xc8202203a0)
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:88 +0x60
created by github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:89 +0x70
FAIL	github.com/cockroachdb/cockroach/storage	300.060s
=== RUN   TestBatchBasics
I1214 22:56:16.301134 1029 storage/engine/rocksdb.go:138  closing in-memory rocksdb instance
--- PASS: TestBatchBasics (0.00s)
=== RUN   TestBatchGet
I1214 22:56:16.304386 1029 storage/engine/rocksdb.go:138  closing in-memory rocksdb instance
--- PASS: TestBatchGet (0.00s)
=== RUN   TestBatchMerge
I1214 22:56:16.309854 1029 storage/engine/rocksdb.go:138  closing in-memory rocksdb instance
--- PASS: TestBatchMerge (0.00s)
=== RUN   TestBatchProto

```
Please assign, take a look and update the issue accordingly.

--- a/storage/scanner.go
+++ b/storage/scanner.go
@@ -208,15 +208,15 @@ func (rs *replicaScanner) scanLoop(clock *hlc.Clock, stopper *stop.Stopper) {
 
 		for {
 			var shouldStop bool
-			if rs.replicas.EstimatedCount() == 0 {
-				// Just wait without processing any replica.
+			count := 0
+			rs.replicas.Visit(func(repl *Replica) bool {
+				count++
+				shouldStop = rs.waitAndProcess(start, clock, stopper, repl)
+				return !shouldStop
+			})
+			if count == 0 {
+				// No replicas processed, just wait.
 				shouldStop = rs.waitAndProcess(start, clock, stopper, nil)
-			} else {
-				shouldStop = true
-				rs.replicas.Visit(func(repl *Replica) bool {
-					shouldStop = rs.waitAndProcess(start, clock, stopper, repl)
-					return !shouldStop
-				})
 			}
 
 			shouldStop = shouldStop || !stopper.RunTask(func() {

